pkg/asset/installconfig/alibabacloud/validation.go

```diff
@@ -117,7 +117,7 @@ func validateClusterName(client *Client, ic *types.InstallConfig) field.ErrorLis
 		allErrs = append(allErrs, field.InternalError(namePath, err))
 	}
 	if response.TotalItems > 0 {
-		allErrs = append(allErrs, field.Invalid(namePath, ic.ClusterName, fmt.Sprintf("cluster name is unavailable, private zone name %s already exists", zoneName)))
+		allErrs = append(allErrs, field.Invalid(namePath, ic.ObjectMeta.Name, fmt.Sprintf("cluster name is unavailable, private zone name %s already exists", zoneName)))
```

Comment:
```suggestion
		allErrs = append(allErrs, field.Invalid(namePath, ic.Name, fmt.Sprintf("cluster name is unavailable, private zone name %s already exists", zoneName)))
```

Reply:
> While we are here fixing field errors, can we fix the typo in namePath where it uses "matedata" instead of "metadata"?

Thanks for reminding！

Sorry, I did not understand this suggestion, the type `types.InstallConfig` does not exist `Name`, or modify `validateClusterName(client *Client, ic *types.InstallConfig)` to `validateClusterName(client *Client, ic *ObjectMeta) `?

Reply:
I remember, because `ic.ObjectMeta.Name` and `ic.Platform.Name` exist at the same time, so `ic.Name` will cause "ambiguous selector" error.